pageserver/src/tenant/upload_queue.rs

```diff
@@ -290,7 +293,10 @@ pub(crate) enum UploadOp {
     Delete(Delete),
 
     /// Barrier. When the barrier operation is reached, the channel is closed.
-    Barrier(tokio::sync::watch::Sender<()>),
+    /// The boolean value indicates whether the barrier is an initial barrier scheduled
+    /// at timeline load -- if yes, we will need to wait for all deletions to be completed
+    /// before the next upload.
+    Barrier(tokio::sync::watch::Sender<()>, bool),
```

Comment:
Let's use a descriptive `enum` rather than a bool here, to make it easier to read & harder to typo